Assets/Mirror/Runtime/NetworkManagerHUD.cs

```diff
@@ -154,5 +228,60 @@ void StopButtons()
                 }
             }
         }
+
+        void SetIPInformation()
+        {
+            if (_IPType == IPType.localhost)
+            {
+                inputField = "localhost";
+                manager.networkAddress = inputField;
+            }
+            else if (_IPType == IPType.LAN)
+            {
+                StartCoroutine(GetLanAddress());
+            }
+            else if (_IPType == IPType.WAN)
+            {
+                StartCoroutine(GetWanAddress());
+            }
+        }
+
+        IEnumerator GetLanAddress()
```

Comment:
why is this a coroutine? that's so complicated.
can't we just have a simple GetLocalAddress() function that returns local address?